Staging fails when files are renamed during commit generation

### Description
I encountered a critical issue where the application fails to stage files when the changeset involves file renames. The tool attempts to stage the original filename rather than the new filename, causing the git operation to fail with a fatal error.

### Reproduction Steps
1. Rename a file in the repository (e.g., `mv internal/agent/bootstrap/s3_reporter.go internal/agent/bootstrap/storage_reporter.go`).
2. Run the commit generation tool.
3. Select the suggested commit message that includes the rename.
4. Confirm the commit.

### Expected Behavior
I expected the tool to correctly identify the file rename and stage the new filename (or handle the index update) so that the commit process completes successfully.

### Actual Behavior
The CLI crashed with a fatal git error because it tried to add the old filename which no longer exists on the filesystem.

Error:
`[ERROR] failed to stage file file=internal/agent/bootstrap/s3_reporter.go error=GIT: Failed to add file to staging (exit status 128) - fatal: specified path 'internal/agent/bootstrap/s3_reporter.go' did not match any files`

### Version
Latest

### Additional Context
Logs from the failure:

```text
📊 Changes

internal/agent/bootstrap/{s3_reporter.go => storage_reporter.go} | 0

.../agent/bootstrap/{s3_reporter_test.go => storage_reporter_test.go} | 0 
2 files changed, 0 insertions(+), 0 deletions(-)

Confirm commit? (y/n): y
⠋ 🧉 Adding files to staging...[ERROR] failed to stage file file=internal/agent/bootstrap/s3_reporter.go error=GIT: Failed to add file to staging (exit status 128) - fatal: specified path 'internal/agent/bootstrap/s3_reporter.go' did not match any files
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -59,4 +59,4 @@ jobs:
         uses: codecov/codecov-action@v5
         with:
           token: ${{ secrets.CODECOV_ORG_TOKEN }}
-          fail_ci_if_error: true
+          fail_ci_if_error: false

--- a/.github/workflows/golangci-lint.yml
+++ b/.github/workflows/golangci-lint.yml
@@ -19,4 +19,4 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v6
         with:
-          version: v1.60
+          version: latest

--- a/internal/ai/gemini/pull_requests_summarizer_service.go
+++ b/internal/ai/gemini/pull_requests_summarizer_service.go
@@ -40,7 +40,7 @@ func getPRSummarySchema() *genai.Schema {
 			},
 			"body": {
 				Type:        genai.TypeString,
-				Description: "Detailed markdown body with overview, key changes, and technical impact",
+				Description: "Detailed PR body as valid Markdown. Use real newline characters between headers, paragraphs and list items (e.g. \"## Title\\n\\nParagraph text\\n\\n- [ ] item\"). Never collapse sections into a single line of prose.",
 			},
 			"labels": {
 				Type: genai.TypeArray,

--- a/internal/ai/prompts.go
+++ b/internal/ai/prompts.go
@@ -74,6 +74,7 @@ const (
   # Golden Rules (Constraints)
   1. **No Hallucinations:** If it's not in the diff, DO NOT invent it.
   2. **Tone:** Professional, direct, technical. Use first person ("I implemented", "I added").
+  3. **Markdown Format (STRICT):** The "body" field MUST be valid Markdown with REAL newline characters (\n) between headers, paragraphs, and list items. Each "## Header" goes on its own line followed by a blank line. Each checklist item ("- [ ] ...") goes on its own line. NEVER merge multiple sections or checklist items into a single paragraph of running text.
   # Instructions
   1. Title: Catchy but descriptive (max 80 chars).
   2. Key Changes: Filter the noise. Explain the *technical impact*, not just the code change.
@@ -86,11 +87,12 @@ const (
   # Reglas de Oro (Constraints)
   1. **Cero alucinaciones:** Si algo no está explícito en el diff, no lo inventes.
   2. **Tono:** Profesional, cercano y directo. Usa primera persona ("Implementé", "Agregué", "Corregí"). Evita el lenguaje robótico ("Se ha realizado").
+  3. **Formato Markdown (ESTRICTO):** El campo "body" DEBE ser Markdown válido con saltos de línea reales (\n) entre encabezados, párrafos e items de lista. Cada "## Encabezado" va en su propia línea seguido de una línea en blanco. Cada checkbox ("- [ ] ...") va en su propia línea. NUNCA mezcles varias secciones o checkboxes en un solo párrafo de texto corrido.
   # Instrucciones
   1. Título: Descriptivo y conciso (máx 80 caracteres).
   2. Cambios Clave: Filtrá el ruido. Explicá el *impacto* técnico y el propósito, no solo qué línea cambió.
   3. Etiquetas: Elegí con criterio (feature, fix, refactor, docs, infra, test, breaking-change).
-  
+
   IMPORTANTE: Responde en ESPAÑOL. Todo el contenido del JSON debe estar en español.`
 )
 

--- a/internal/git/git_service.go
+++ b/internal/git/git_service.go
@@ -79,6 +79,12 @@ func (s *GitService) GetChangedFiles(ctx context.Context) ([]string, error) {
 		if len(line) > 3 {
 			path := strings.TrimSpace(line[3:])
 
+			// Rename/copy entries are reported as "old -> new"; only the
+			// new path exists on disk and is relevant for staging/diffing.
+			if idx := strings.Index(path, " -> "); idx != -1 {
+				path = path[idx+len(" -> "):]
+			}
+
 			if path != "" {
 				changes = append(changes, path)
 			}

--- a/internal/git/git_service_test.go
+++ b/internal/git/git_service_test.go
@@ -157,6 +157,54 @@ func TestGitService(t *testing.T) {
 		}
 	})
 
+	t.Run("GetChangedFiles con archivo renombrado", func(t *testing.T) {
+		// arrange
+		tempDir := setupTestRepo(t)
+		defer cleanupTestRepo(t, tempDir)
+
+		if err := os.WriteFile("old.txt", []byte("contenido"), 0644); err != nil {
+			t.Fatalf("Error creando archivo original: %v", err)
+		}
+		if err := exec.Command("git", "add", "old.txt").Run(); err != nil {
+			t.Fatalf("Error agregando archivo original: %v", err)
+		}
+		if err := exec.Command("git", "commit", "-m", "init").Run(); err != nil {
+			t.Fatalf("Error creando commit inicial: %v", err)
+		}
+		if err := os.Rename("old.txt", "new.txt"); err != nil {
+			t.Fatalf("Error renombrando archivo: %v", err)
+		}
+		if err := exec.Command("git", "add", "-A").Run(); err != nil {
+			t.Fatalf("Error agregando cambios: %v", err)
+		}
+
+		service := NewGitService()
+
+		// act
+		changes, err := service.GetChangedFiles(context.Background())
+
+		// assert
+		if err != nil {
+			t.Errorf("Error obteniendo archivos cambiados: %v", err)
+		}
+
+		for _, path := range changes {
+			if strings.Contains(path, " -> ") {
+				t.Errorf("el path no debe contener la notación de rename cruda: %q", path)
+			}
+		}
+
+		found := false
+		for _, path := range changes {
+			if path == "new.txt" {
+				found = true
+			}
+		}
+		if !found {
+			t.Errorf("se esperaba encontrar 'new.txt' entre los cambios, se obtuvo: %v", changes)
+		}
+	})
+
 	t.Run("CreateCommit", func(t *testing.T) {
 		// arrange
 		tempDir := setupTestRepo(t)

